Script for model analysis and store information in  markdown file

chandrasekaranpradeep opened the pull request:
Fixes #661 
1. Add option for exporting model unique op configuration to the excel file
2. Created a script for generating and exporting unique op configuration test and run the unique op test and generate markdown files, the root markdown file contains model name, variant name, framework and compiler components supported rate and sub-markdown file contains  model variant unique op tests info

Generated Markdown file for preview - [ModelsInfo](https://github.com/tenstorrent/tt-forge-fe/blob/pchandrasekaran/model_preview/models_analysis_docs/ModelsInfo.md)

chandrasekaranpradeep commented:
FYI, [model_analysis_docs](https://github.com/tenstorrent/tt-forge-fe/tree/pchandrasekaran/model_analysis_markdown_gen/models_analysis_docs) folder will removed from the PR before merging into main.

nvukobratTT commented on scripts/model_analysis.py:
```diff
@@ -0,0 +1,887 @@
+        return None, None
+
+
+common_failure_matching_rules_list = [
```
When we start with refactor, let's put these error mappings in separated file for easier update and manage over time

chandrasekaranpradeep replied:
Sure Nikola I will move it to separate file while restructing

chandrasekaranpradeep replied:
Add the task of moving  error mappings in separated file to the restructure issues - https://github.com/tenstorrent/tt-forge-fe/issues/723

nvukobratTT commented on scripts/model_analysis.py:
```diff
@@ -0,0 +1,887 @@
+    model_output_dir_paths, markdown_directory_path, dump_failure_logs
+):
+    """
+    Execute unique operation tests for specified models, gather compiler support details and
```
One of the future optimizations can be combine all unique tests, run them and populate model information. Running unique op tests per model will for sure add redundancy. However, let's have a separate issues for this optimization once functional logic is in place.

chandrasekaranpradeep replied:
Instead of running unique op test by model, we should collect the unique op tests from all the models and avoid running the duplicate unique op test, for example if we have add op with input shape (1, 32) and (1, 32) in resnet and yolo, we should run only once and populate the test details for both yolo and resnet.
Am I right? @nvukobratTT

chandrasekaranpradeep replied:
> Yup, that's correct :)) I'm certain there are many ops with sam input shape and configurations repeating. If we only run unique ones, we'll get an additional boost. However, this should be additional optimization.

Okay Nikola

chandrasekaranpradeep replied:
Created a separate issue for optimizing the model analysis script and generate nightly test - https://github.com/tenstorrent/tt-forge-fe/issues/761

chandrasekaranpradeep commented:
@nvukobratTT  Update the model_analysis script to include bit more error message in compiler component mapper (i.e common_failure_matching_rules_list), added some check and assert statement in the script and created HtmlSymbol enum for defining the html symbols and added last updated data and time column to the modelsinfo.md.

Here is the final generated model analysis md file preview - [link](https://github.com/tenstorrent/tt-forge-fe/blob/pchandrasekaran/model_preview/new_models_analysis_docs/ModelsInfo.md)
